src/hooks/api/user.js

```diff
@@ -1,26 +1,64 @@
 import { fetchWithRedirect } from '../../utils/fetchWithRedirect';
 import { useCookiesWithOptions } from '../useCookiesWithOptions';
 import { useAuthenticatedQuery } from './useAuthenticatedQuery';
+import { useHeaders } from './useHeaders';
 
-const getMe = async (token) => {
+const getMe = async (headers) => {
   const res = await fetchWithRedirect(
     `${process.env.NEXT_PUBLIC_API_URL}/user`,
     {
-      headers: {
-        Authorization: `Bearer ${token}`,
-        'X-Api-Key': process.env.NEXT_PUBLIC_API_KEY,
-      },
+      headers,
     },
   );
   return await res.json();
 };
 
 const useGetUser = (config = {}) => {
   const [cookies] = useCookiesWithOptions(['token']);
-  return useAuthenticatedQuery('user', () => getMe(cookies.token), {
+  const headers = useHeaders();
+  return useAuthenticatedQuery('getUser', () => getMe(headers), {
     enabled: cookies.token,
     ...config,
   });
 };
 
-export { useGetUser };
+const getPermissions = async (headers) => {
+  const res = await fetchWithRedirect(
+    `${process.env.NEXT_PUBLIC_API_URL}/user/permissions/`,
+    {
+      headers,
+    },
+  );
+  return await res.json();
+};
+
+const useGetPermissions = (config = {}) => {
+  const [cookies] = useCookiesWithOptions(['token']);
+  const headers = useHeaders();
+  return useAuthenticatedQuery(
+    'getPermissions',
+    () => getPermissions(headers),
+    {
+      enabled: cookies.token,
+      ...config,
+    },
+  );
+};
+
+const getRoles = async (headers) => {
+  const res = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/user/roles/`, {
+    headers,
+  });
+  return await res.json();
+};
+
+const useGetRoles = (config = {}) => {
+  const [cookies] = useCookiesWithOptions(['token']);
+  const headers = useHeaders();
```

Comment:
I'm wondering if we could move this inside for example `fetchWithRedirect` or `useAuthenticatedQuery` so we don't have to repeat this for every API request hook.
(Looking at the name `useAuthenticatedQuery` I'd also assume at first glance that that would add the necessary authentication headers)

Reply:
See changes I've made in [this commit](https://github.com/cultuurnet/udb3-frontend/pull/124/commits/473e5060576d86889700a6581fe4b1c68a9e16bc)

Reply:
Looks great!